src/zarr/v3/abc/codec.py

```diff
@@ -16,6 +25,44 @@
     from zarr.v3.metadata import ArrayMetadata
     from zarr.v3.config import RuntimeConfiguration
 
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+def noop_for_none(
+    func: Callable[[T, ArraySpec, RuntimeConfiguration], Awaitable[Optional[U]]],
```

Comment:
I'm having trouble understanding why this function exists just from its signature -- could you add a docstring that gives some context for it?

Reply:
I think I'm also struggling to understand the role of `None` in the chunk encoding / decoding process

Reply:
Since this is the batch mode, we're dealing with multiple chunk requests. If the store returns `None` (i.e. missing chunk = fill value-only chunk) for any of these, we don't try to decode these. The `noop_for_none` function is just a higher order function so that the decode (or encode) functions don't have to do the early return on their own.

Reply:
Do you think we could simplify the logic here if stores didn't return `None`?

Reply:
Not really. In that case, the codec pipeline would have to handle KeyError or similar.

Reply:
> Not really. In that case, the codec pipeline would have to handle KeyError or similar.

But if handling the key error was in just one place, and after handling keyerror the empty keys were turned into arrays of `fill_value` (or just ignored completely, if there was already a buffer pre-filled with `fill_value`), then I think there would be no need to handle `None` in any of the chunk processing logic, which would be kind of nice

Reply:
I might be wrong but handling a `KeyError` or a `None` is the same complexity in the chunk processing logic. Both still have to happen per-chunk and coordinated with the other chunks in a batch request.

Reply:
I wonder if missing chunks should never enter the encoding / decoding process at all?

if we handle missing chunks early (e.g., right when we fetch them from the store, for decoding, or when we fetch them from the array, for encoding), then the chunk encoding / decoding routines never need to handle missing chunks, which means that `noop_for_none` wouldn't need to exist, because `chunk` could never be `None`, and likewise for all the other methods that take `Optional[np.ndarray]` as an argument.

Reply:
The codec pipeline does the fetching so that code needs to live there. It also merges chunks into the decoded ndarray, for that it also needs to know which chunks are missing to fill them with fill_value. Using None to propagate that information seems ok.